spec/src/main/asciidoc/configsources.asciidoc

```diff
@@ -24,7 +24,8 @@
 == ConfigSources
 
 A `ConfigSource` is exactly what its name says: a source for configured values.
-The `Config` uses all configured implementations of `ConfigSource` to look up the property in question.
+The `Config` uses all configured implementations of `ConfigSource` to look up the property in question. Dynamically added config sources after the `Config` object has been built would be ignored, 
+which means `Config.getConfigSources` always returns the same collection of `ConfigSource`s. The same rule applies to `ConfigSourceProvider.getConfigSources`.
 
```

Comment:
Did [this comment](https://github.com/eclipse/microprofile-config/issues/431#issuecomment-519858099) ever make it into the spec?  If not, it would probably be best to put it here.

Reply:
@ljnelson it is pretty much common sense implementation. I prefer not to put it here as getPropertyNames() could give you a different list depending when you call it since some configsources are dynamic. This issue is about we ensure the number of configsources are determined and not dynamic while the config sources might be dynamic.

Reply:
IMHO specifications should *never ever* rely on "common sense".  "Common sense" is "undefined behavior".

Here, it is best to specify what the behavior of `Config#getPropertyNames()` is with respect to the behavior of `ConfigSource#getPropertyNames()`, because it does not seem to be specified anywhere at the moment.  If you do not want to specify it, then an explicit statement to that effect would be important I think.